src/main/scala/fs2/kafka/KafkaAdminClient.scala

```diff
@@ -35,6 +35,21 @@ import org.apache.kafka.common.{KafkaFuture, TopicPartition}
   */
 sealed abstract class KafkaAdminClient[F[_]] {
 
+  /**
+    * Create a new topic.
+    *
+    * NOTE: Kafka's base [[AdminClient]] supports creating multiple topics in a
+    * single request, but the result isn't transactional. We limit to a single
+    * topic-per-request to avoid complications in error reporting for cases when
+    * some topic creations fail and others succeed.
```

Comment:
How do you all feel about this design choice? I couldn't think of an obvious way to convert the `Map[String, Future[Void]]` of topic-creation results provided by Kafka into a single `F[_]` capturing all failures and all successes, and figured it's easy enough for users to build that behavior however they'd like using Cats and fs2.

Reply:
We can do something similar to what we do for e.g. `describeTopics`:

```scala
def createTopics[G[_]](topics: G[NewTopic])(
  implicit G: Foldable[G]
): F[Unit]
```

and `createTopicsWith` then simply becomes the following.

```scala
private[this] def createTopicsWith[F[_], G[_]](
  client: Client[F],
  topics: G[NewTopic]
)(implicit G: Foldable[G]): F[Unit] =
  client(_.createTopics(topics.asJava).all.void)
```

You might need to add `void` to `KafkaFutureSyntax` in `fs2.kafka.internal.syntax`.

```scala
def void: KafkaFuture[Unit] =
  map(_ => ())
```

Reply:
:+1: to using `Foldable` to handle the input collection, but my worry is about the `.all` call. The underlying `KafkaFuture.allOf` arbitrarily chooses a single error to bubble up if multiple Futures have failed. For our use-case, we'd need to be able to collect & report every failure.

If there's interest on your end, I can spend some time setting up more complex error-handling (i.e. using a custom `Exception` type that can collect all the errors). Otherwise I'll make the change as you suggest, and we'll just only call the method with one topic at a time.

Reply:
I think it's difficult to cater for all use cases here, but it's pretty easy to write yourself using only `createTopics` (or `createTopic` for a single topic), `parTraverse`, and `attempt`. You can also easily include the topic name, or other details you care about.

```scala
topics.parTraverse(topic => client.createTopic(topic).attempt)
// e.g. IO[List[Either[Throwable,Unit]]]
```

I've pushed some changes, including `createTopic`. Would you mind having a look?